### ISSUE TITLE
`connect` reruns queries after store changes

### ISSUE THREAD

**Issue by sgoll** [apollostack/react-apollo#90]
I stumbled upon an intricate bug, I think.

Whenever the Redux store is changed (by some arbitrary reducer outside of Apollo), `connect` calls `mapQueriesToProps` and checks if the returned queries have changed, re-evaluating them if necessary: https://github.com/apollostack/react-apollo/blob/master/src/connect.tsx#L263

So far, so good. However, the `isEqual` check here returns false whenever a query returned by `mapQueriesToProps` does not specify a `forceFetch` option. The reason for this is that the objects compared here at the time of comparison look something like this:

```
oldQueries = {
  someQuery: {
    forceFetch: undefined,
    query: (…),  // query
    variables: (…),  // variables
  },
}

queryHandles = {
  someQuery: {
    query: (…),  // query
    variables: (…),  // variables
  },
}
```

In other words, the original queryHandles saved into this.previousQueries and available here as `oldQueries` gets modified by having `forceFetch` attached which in turn makes it unequal when the same `queryHandles` is passed into connect later on.

The location where the setting of `forceFetch` happens is in apollo-client, in the `watchQuery`/`query` wrappers: https://github.com/apollostack/apollo-client/blob/master/src/index.ts#L119

Here, `options.forceFetch` is explicitly set (possibly to undefined) which in turn modifies the shallowly-copied `oldQueries` above.

As this bug concerns both apollo-client and react-apollo, I am not sure how best to proceed here. Should it be fixed (or worked around?) in one or the other package?

I guess, my hunch would be to replace the line concerning shouldForceFetch in apollo-client with something like this:

```
const forceFetch = this.shouldForceFetch && options.forceFetch;
if (forceFetch !== options.forceFetch) {
  // Don't modify passed-in `options` object.
  options = assign({}, options, { forceFetch });
}
```

**Comment by stubailo** [comment]
Fixed and released in `0.3.25`!

Thank you for the extremely detailed analysis, I hope that fix works. Please reopen if it doesn't.

**Comment by sgoll** [comment]
@stubailo Perfect solution, thank you. The fix works as intended.
